src/main/java/au/org/ala/biocache/service/DownloadService.java

```diff
@@ -402,214 +553,326 @@ public void getCitations(Map<String, Integer> uidStats, OutputStream out, char s
-                            });
+                            writer.writeNext(new String[] { headerOutput[i], fieldsRequested[i], "", "", "", "", "",
+                                    info != null ? info : "" });
                         }
                     }
                 }
-    
-                //misc headers
+
+                // misc headers
                 if (miscHeaders != null) {
                     for (int i = 0; i < miscHeaders.length; i++) {
-                        writer.writeNext(new String[]{miscHeaders[i],
-                                "",
-                                "",
-                                "",
-                                "",
-                                "",
-                                "Raw header from data provider."
-                        });
+                        writer.writeNext(
+                                new String[] { miscHeaders[i], "", "", "", "", "", "Raw header from data provider." });
                     }
                 }
-    
+
                 writer.flush();
             }
         }
     }
 
-    private String getOrElse(Map map, String key, String value){
-        if(map.containsKey(key)){
+    private String getOrElse(Map<String, Object> map, String key, String value) {
+        if (map.containsKey(key)) {
             return map.get(key).toString();
-        } else{
+        } else {
             return value;
         }
     }
-    
+
     /**
-     * A thread responsible for creating a records dump offline.
+     * A thread to abstract away the details of the ExecutorService/Callable being used. 
      * 
-     * @author Natasha Carter (natasha.carter@csiro.au)
+     * @author Peter Ansell
      */
-    private class DownloadThread implements Runnable {
-        
-        private DownloadDetailsDTO currentDownload = null;
+    private class DownloadServiceExecutor {
 
-        private Integer maxRecords = null;
-        private DownloadType downloadType = null;
+        private final Integer maxRecords;
+        private final DownloadType downloadType;
+        private final ExecutorService executor;
+        private final long executionDelay;
+        private final int priority;
 
-        public DownloadThread(Integer maxRecords, DownloadType downloadType) {
+        public DownloadServiceExecutor(Integer maxRecords, DownloadType downloadType, int concurrencyLevel) {
             this.maxRecords = maxRecords;
             this.downloadType = downloadType;
+            if (maxRecords == null || maxRecords > 50000) {
+                executionDelay = 100 + Math.round(Math.random() * 200);
+                // Large downloads receive minimum thread priority to give lower latency on short downloads
+                priority = Thread.MIN_PRIORITY;
+            } else {
+                executionDelay = 1 + Math.round(Math.random() * 10);
+                priority = Thread.NORM_PRIORITY;
+            }
```

Comment:
Change static definitions of `executionDelay` and `priority` to be configurable in `concurrentDownloadsExtra` https://github.com/AtlasOfLivingAustralia/biocache-service/blob/9b4cba0db01f078a6f6acde00ae0eab87d71cee1/src/main/java/au/org/ala/biocache/service/DownloadService.java#L70